LiskHQ/lisk-mobile
# You can't run the app in discreet mode if it's already run - Closes #887

# What was the bug or feature?
Described in #887.

### How did I fix it?
The action to activate discreet mode is dispatched from `SignIn` whenever the discreet quick action is triggered. That way it is guaranteed that the navigation to home is always done in discreet mode, even when the `Home` component is already mounted or the app is already running. If discreet mode is already active, the `Home` component will not dispatch the discreet mode action. 
If the active screen is `Register`, `popToTop` is called to go back to `SignIn`. After successful login, navigation to `Home` is done discreetly.

## Type of change
- Bug fix (a non-breaking change which fixes an issue)

### How to test it?
Consider the following points:
- If the app is not run, and user opens in discreet mode, it should open the sign in page and after successful sign in, the home page must render in discreet mode.
- If the app is run and in sign in page, successful authentication should navigate the user to home page in discreet mode.
- If the app is signed in and minimised, maximising using the open discreetly menu item
must only turn the discreet mode on and not change the route.
- If the app is in registration page, using the open discreetly menu item must navigate to the sign in page and allow user to sign-in in discreet mode.


# Checklist:
- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
